src/states/node.rs

```diff
@@ -971,63 +1033,134 @@ impl Node {
                         debug!("{:?} Prepared connection info for {:?}.",
                                self,
                                pub_id.name());
+                        if let Some(our_info) = self.peer_mgr.our_connection_info(pub_id.name()) {
```

Comment:
Why not just call `our_connection_info.to_pub_connection_info()` above? Then this method, and in particular the unreachable `error!`, wouldn't be necessary.